Update Geth installation page (#24777)
ethereum/go-ethereum: docs/_install-and-build/Installing-Geth.md

{:toc}

-   this will be removed by the toc

## Package managers

### MacOS via Homebrew

The easiest way to install go-ethereum is to use the Geth Homebrew tap. The first step is to check that Homebrew is installed. The following command should return a version number.

<br>
```shell
brew -v
```
<br>

If a version number is returned, then Homebrew is installed. If not, Homebrew can be installed by following the instructions [here][brew]. With Homebrew installed, the following commands add the Geth tap and install Geth:

<br>
```shell
brew tap ethereum/ethereum
brew install ethereum
```
<br>

The previous command installs the latest stable release. Developers that wish to install the most up-to-date version can install the Geth repository's master branch by adding the `--devel` parameter to the install command:

<br>
```shell
brew install ethereum --devel
```
<br>

> These commands install the core Geth software and the following developer tools: `clef`, `devp2p`, `abigen`, `bootnode`, `evm`, `rlpdump` and `puppeth`. The binaries for each of these tools are saved in `/usr/local/bin/`. The full list of command line options can be viewed in the terminal by running `geth --help`.

Reviewer: Just out of curiosity, where did you get this list? check the binaries installed?

Author: This list came from the Executables section in the main Geth repository (which I assume to be more up to date than the website docs): https://github.com/ethereum/go-ethereum#executables

> These commands install the core Geth software and the following developer tools: `clef`, `devp2p`, `abigen`, `bootnode`, `evm`, `rlpdump` and `puppeth`. The binaries for each of these tools are saved in `/usr/local/bin/`. The full list of command line options can be viewed in the terminal by running `geth --help`.

Reviewer: Also command line options could be linked to the corresponding page

Author: yep great, will add this.

Updating an existing Geth installation to the latest version can be achieved by stopping the node and running the following commands:

<br>
```shell
brew update 
brew upgrade 
brew reinstall ethereum
```
<br>

When the node is started again, Geth will automatically use all the data from the previous version and sync the blocks that were missed while the node was offline.

### Ubuntu via PPAs

> The easiest way to install Geth on Ubuntu-based distributions is with the built-in launchpad PPAs (Personal Package Archives). A single PPA repository is provided, containing stable and development releases for Ubuntu versions `trusty`, `xenial`, `zesty` and `artful`.

Reviewer: The ubuntu versions seem to be outdated. From the [PPA](https://launchpad.net/~ethereum/+archive/ubuntu/ethereum) page:

xenial, trusty, impish, focal, bionic

Author: thanks for the catch, updating now.

The following command enables the launchpad repository:

<br>
```shell
sudo add-apt-repository -y ppa:ethereum/ethereum
```
<br>

Then, to install the stable version of go-ethereum:

<br>
```shell
sudo apt-get update
sudo apt-get install ethereum
```
<br>

Or, alternatively the develop version:

<br>
```shell
sudo apt-get update
sudo apt-get install ethereum-unstable
```
<br>

These commands install the core Geth software and the following developer tools: `clef`, `devp2p`, `abigen`, `bootnode`, `evm`, `rlpdump` and `puppeth`. The binaries for each of these tools are saved in `/usr/local/bin/`. The full list of command line options can be viewed in the terminal by running `geth --help`.

<br>
Updating an existing Geth installation to the latest version can be achieved by stopping the node and running the following commands:

<br>
```shell
sudo apt-get update
sudo apt-get install ethereum
sudo apt-get upgrade geth
```
<br>

When the node is started again, Geth will automatically use all the data from the previous version and sync the blocks that were missed while the node was offline.

### Windows

> The easiest way to install go-ethereum is to download a pre-compiled binary from the [downloads][geth-dl] page. The page provides an installer as well as a zip file containing the Geth source code. The install wizard offers the user the option to install Geth, or Geth and the developer tools. The installer adds `geth` to the system's `PATH` automatically. The zip file contains the command `.exe` files that can be run from the command prompt. The full list of command line options can be viewed in the terminal by running `geth --help`.

Reviewer: ```suggestion
The easiest way to install Geth is to download a pre-compiled binary from the [downloads][geth-dl] page. The page provides an installer as well as a zip file containing the Geth source code. The install wizard offers the user the option to install Geth, or Geth and the developer tools. The installer adds `geth` to the system's `PATH` automatically. The zip file contains the command `.exe` files that can be run from the command prompt. The full list of command line options can be viewed in the terminal by running `geth --help`.
```

Updating an existing Geth installation can be achieved by stopping the node, downloading and installing the latest version followign the instructions above. When the node is started again, Geth will automatically use all the data from the previous version and sync the blocks that were missed while the node was offline.

### FreeBSD via pkg

Geth can be installed on FreeBSD using the package manager `pkg`. The following command downloads and installs Geth:

<br>
```shell
pkg install go-ethereum
```
<br>

These commands install the core Geth software and the following developer tools: `clef`, `devp2p`, `abigen`, `bootnode`, `evm`, `rlpdump` and `puppeth`. 

The full list of command line options can be viewed in the terminal by running `geth --help`.

Updating an existing Geth installation to the latest version can be achieved by stopping the node and running the following commands:

<br>
```shell
pkg upgrade
```
<br>

When the node is started again, Geth will automatically use all the data from the previous version and sync the blocks that were missed while the node was offline.

### FreeBSD via ports

> Installing Geth using ports, simpyl requires navigating to the `net-p2p/go-ethereum` ports directory and running `make install` as root:

Reviewer: ```suggestion
Installing Geth using ports, simply requires navigating to the `net-p2p/go-ethereum` ports directory and running `make install` as root:
```

-   `ethereum/client-go:alltools-{version}` is the stable version of the Ethereum tools at a specific version number
-   `ethereum/client-go:alltools-release-{version}` is the latest stable version of the Ethereum tools at a specific version family
<br>
<br>

The image has the following ports automatically exposed:
<br>
<br>

-   `8545` TCP, used by the HTTP based JSON RPC API
-   `8546` TCP, used by the WebSocket based JSON RPC API
-   `8547` TCP, used by the GraphQL API
-   `30303` TCP and UDP, used by the P2P protocol running the network
<br>
<br>

**Note:** if you are running an Ethereum client inside a Docker container, you should mount a data volume as the client's data directory (located at `/root/.ethereum` inside the container) to ensure that downloaded data is preserved between restarts and/or container life-cycles.

<br>
Updating Geth to the latest version simply requires stopping the container, pulling the latest version from Docker and running it:

<br>
```shell
docker stop ethereum/client-go
docker pull ethereum/client-go:latest
docker run -it -p 30303:30303 ethereum/client-go
```
<br>

## Build from source code

### Most Linux systems and macOS

> Go Ethereum is written in [Go][go], so to building from source code requires the most recent version of Go to be installed. Instructions for installing GO are available at the [Go installation page][go-install] and necessary bundles can be downlaoded from the [Go download page][go-dl].

Reviewer: ```suggestion
Geth is written in [Go][go], so building from source code requires the most recent version of Go to be installed. Instructions for installing Go are available at the [Go installation page][go-install] and necessary bundles can be downloaded from the [Go download page][go-dl].
```
